spock-specs/src/test/groovy/org/spockframework/docs/extension/TempDirDocSpec.groovy

```diff
@@ -0,0 +1,49 @@
+package org.spockframework.docs.extension
+
+import spock.lang.Shared
+import spock.lang.Specification
+import spock.lang.TempDir
+
+import java.nio.file.Path
+
+/**
+ * @author dqyuan
+ * @since 2.0
+ */
+class TempDirDocSpec extends Specification {
+
+  def placeholder() {
+    expect: true
+  }
+
+// tag::example-a[]
+  // all feature will share the same temp directory path1
+  @TempDir
+  @Shared
+  Path path1
+
+  // every iteration will have its own path2
+  @TempDir
+  File path2
+
+  // will be injected using java.nio.file.Path
+  @TempDir
+  def path3
+// end::example-a[]
+}
+
+
+class TempDirParamSpec extends Specification {
```

Comment:
`example-b` is obsolete, so this whole second class can be deleted and `example-a` renamed to `example`